segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java

```diff
@@ -172,13 +172,14 @@ public void close() {
     public void insert(Key key, byte[] data) {
         ensureInitializedAndNotClosed();
         Timer timer = new Timer();
+        byte[] serializedKey = key.serialize();
         try {
-            this.database.get().put(this.writeOptions, key.serialize(), data);
+            this.database.get().put(this.writeOptions, serializedKey, data);
         } catch (RocksDBException ex) {
             throw convert(ex, "insert key '%s'", key);
         }
 
-        RocksDBMetrics.insert(timer.getElapsedMillis());
+        RocksDBMetrics.insert(timer.getElapsedMillis(), serializedKey.length + data.length);
```

Comment:
Can `data` be null? I doubt it, but should we check?

Reply:
It does not seem so according to the static analysis of IntelliJ, but added the check anyway as we already do for `get`s.